### PR TITLE
@alpaca/liquidation alert

### PR DESCRIPTION
**Pull request by RCantu92**
`ALPACA-3` agent detects `Kill` event emissions from `Vault` contracts.
`ALPACA-4` agent detects `Kill` event emission from `Vault` contracts when `left` equals zero.

### CODE REVIEW

File: Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+const createFindings = (
+  posId: number,
+  killer: string,
+  posOwner: string,
+  posVal: number,
+  debt: number,
+  prize: number,
+  left: number,
+  vaultAddress: string
+): Finding[] => {
```
Comment (mauricio1802) [inline Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts:29]: I think you shouldn't delegate the responsibility for deciding which alerts to emit to this function.

File: Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+export function provideHandleTransaction(
+  addresses: string[]
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const killEvents = txEvent.filterLog(killEventAbi);
+
+    for(let i = 0; i < killEvents.length; i++) {
+      for(let v = 0; v < addresses.length; v++) {
```
Comment (mauricio1802) [inline Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts:88]: You have to remove this `for` loop. If you leave that for you would be checking if the address emitting the event is a relevant address `N` times, where `N` is the length of the `addresses` array. 
I think you can add one more address to the `TEST_VAULT_ADDRESSES` for checking what I am saying

**Review by mauricio1802 — CHANGES_REQUESTED** [review]

**Review by mauricio1802 — APPROVED** [review]

File: Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          killEvents[i].args["owner"],
+          killEvents[i].args["posVal"],
+          killEvents[i].args["debt"],
+          killEvents[i].args["prize"],
+          killEvents[i].args["left"],
+          killEvents[i].address
+        );
+        findings.push(newAgentThreeFinding);
+
+        if(Number(killEvents[i].args["left"]) === 0) {
```
Comment (e1Ru1o) [inline Alpaca-Finance-Agents/Liquidation-Alert/src/agent.ts:101]: This seems safer to me:
```suggestion
        if(BigNumber.from(0).eq(killEvents[i].args["left"])) {
```
I am not sure that parsing to `Number` will be safe.

You should import `BigNumber` from `ethers`:
`import { BigNumber } from "ethers";`

**Review by e1Ru1o — CHANGES_REQUESTED** [review]

File: Alpaca-Finance-Agents/Liquidation-Alert/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  - Severity is always set to "Info."
+  - Type is always set to "Info."
+  - Metadata contains:
+    - `positionId`: Id of the target position.
+    - `positionkiller`: Address that executed the liquidation.
+    - `positionOwner`: Address of the owner of the target position.
+    - `positionValue`: Value amount of target position.
+    - `debt`: Debt value of the target position.
+    - `prize`: Combined amount of what is awarded to the liquidator and the treasury fees.
+    - `left`: What is returned to the target position's owner after liquidation execution.
```
[the remark this answers is not in the document]
Comment (RCantu92) [reply Alpaca-Finance-Agents/Liquidation-Alert/README.md:39]: That is true.

However, the other agent would also include it and perhaps including it in the one that confirms it is zero can display it for "peace of mind" for whoever sees it. Thoughts? I'm kind of leaning towards leaving it, especially if it doesn't hurt to include it.

[the remark this answers is not in the document]
Comment (e1Ru1o) [reply Alpaca-Finance-Agents/Liquidation-Alert/README.md:39]: Keep it

**Review by e1Ru1o — APPROVED** [review]

**Pull request merged**